cmd/plc-server/server.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"reflect"
+	"strings"
+
+	"github.com/stellentus/go-plc"
+)
+
+type RawTagsHandler struct {
+	plc.ReadWriter
+	validTags map[string]interface{}
+}
+
+func (h RawTagsHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	switch req.Method {
+	case "GET":
+		h.Read(w, req)
+	case "POST":
+		h.Write(w, req)
+	default:
+		logHTTPError(w, http.StatusBadRequest, "Unexpected verb: "+req.Method)
+	}
+}
+
+func logHTTPError(w http.ResponseWriter, code int, reason string) {
+	log.Printf("Bad http request: %s (%d) - %s\n", http.StatusText(code), code, reason)
+	w.WriteHeader(code)
+	w.Write([]byte(reason))
+}
+
+func copy(v interface{}) interface{} {
```

Comment:
To me, "copy" implies an equivalent copy of `v` is created, but I think this just creates the default value of `v`'s type.  Do we want the returned value to have the same contents (or am I misunderstanding `reflect.New()`'s docstring)?  

If not, maybe the name should be adjusted to suggest it constructs something of v's type but the returned value is distinct in value?  I can't come up with something I really like, but something to the effect of `newTypeOf`, or something, maybe?

Reply:
The name is even slightly more misleading than that: it's actually `newPointerOfType`. It's returning a pointer to an allocated variable of that type, not returning the variable itself.